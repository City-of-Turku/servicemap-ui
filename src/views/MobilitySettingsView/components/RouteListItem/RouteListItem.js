import React from 'react';
import { Checkbox, FormControlLabel, Typography } from '@material-ui/core';
import PropTypes from 'prop-types';

const RouteListItem = ({
  classes,
  item,
  routeAttr,
  type,
  setRouteState,
  getRouteName,
  children,
}) => (
  <div key={item.id} className={classes.checkBoxItem}>
    <FormControlLabel
      control={(
        <Checkbox
          checked={type === 'CultureRoute' ? item.id === routeAttr : item.name_fi === routeAttr}
          aria-checked={type === 'CultureRoute' ? item.id === routeAttr : item.name_fi === routeAttr}
          className={classes.margin}
          onChange={() => setRouteState(type === 'CultureRoute' ? item.id : item.name_fi)}
        />
          )}
      label={(
        <Typography variant="body2" aria-label={getRouteName(item.name_fi, item.name_en, item.name_sv)}>
          {getRouteName(item.name_fi, item.name_en, item.name_sv)}
        </Typography>
          )}
    />
    {children}
  </div>
);

RouteListItem.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  item: PropTypes.objectOf(PropTypes.any),
  routeAttr: PropTypes.string,
  type: PropTypes.string,
  setRouteState: PropTypes.func.isRequired,
  getRouteName: PropTypes.func.isRequired,
  children: PropTypes.node,
};

RouteListItem.defaultProps = {
  item: {},
  routeAttr: '',
  type: '',
  children: null,
};

export default RouteListItem;
