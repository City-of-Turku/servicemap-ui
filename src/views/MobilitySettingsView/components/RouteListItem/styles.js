const styles = theme => ({
  checkBoxItem: {
    borderBottom: '1px solid rgb(193, 193, 193)',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'start',
    paddingLeft: theme.spacing(3.5),
  },
});

export default styles;
