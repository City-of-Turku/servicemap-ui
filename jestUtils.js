import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });