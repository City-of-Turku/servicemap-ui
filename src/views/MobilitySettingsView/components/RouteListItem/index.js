import { withStyles } from '@mui/styles';
import RouteListItem from './RouteListItem';
import styles from './styles';

export default withStyles(styles)(RouteListItem);
