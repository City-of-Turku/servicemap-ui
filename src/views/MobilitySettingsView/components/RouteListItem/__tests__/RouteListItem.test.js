import React from 'react';
import RouteListItem from '../index';
import { getRenderWithProviders } from '../../../../../../jestUtils';
import { initialState } from '../../../../../redux/reducers/user';

const mockProps = {
  item: {
    id: '123test',
    name_fi: 'reitti',
    name_en: 'route',
    name_sv: 'rutt',
  },
  routeAttr: '123test',
  type: 'CultureRoute',
  setRouteState: jest.fn,
};

const renderWithProviders = getRenderWithProviders({
  user: initialState,
});

describe('<RouteListItem />', () => {
  it('should match snapshot', () => {
    const { container } = renderWithProviders(<RouteListItem {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<RouteListItem {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toContain(mockProps.item.name_fi);
  });

  it('does contain aria-label attribute', () => {
    const { container } = renderWithProviders(<RouteListItem {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].getAttribute('aria-label')).toContain(mockProps.item.name_fi);
  });
});
