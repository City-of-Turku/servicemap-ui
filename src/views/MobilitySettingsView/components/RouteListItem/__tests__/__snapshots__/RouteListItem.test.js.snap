// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouteListItem /> should match snapshot 1`] = `
<div>
  <div
    class="RouteListItem-checkBoxItem-1"
  >
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        aria-checked="true"
        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-fy5xgy-MuiButtonBase-root-MuiCheckbox-root"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-6yqi1c-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
      </span>
      <p
        aria-label="reitti"
        class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
      >
        reitti
      </p>
    </label>
  </div>
</div>
`;
